crates/consensus-logic/src/csm/worker.rs

```diff
@@ -0,0 +1,372 @@
+//! Consensus logic worker task.
+
+// TODO massively refactor this module
+
+use std::{sync::Arc, thread};
+
+use strata_db::{
+    traits::*,
+    types::{CheckpointConfStatus, CheckpointEntry, CheckpointProvingStatus},
+};
+use strata_eectl::engine::ExecEngineCtl;
+use strata_primitives::prelude::*;
+use strata_state::{client_state::ClientState, csm_status::CsmStatus, operation::SyncAction};
+use strata_status::StatusTx;
+use strata_storage::{managers::checkpoint::CheckpointDbManager, L2BlockManager};
+use strata_tasks::ShutdownGuard;
+use tokio::{
+    sync::{broadcast, mpsc},
+    time,
+};
+use tracing::*;
+
+use super::{
+    config::CsmExecConfig,
+    message::{ClientUpdateNotif, CsmMessage, ForkChoiceMessage},
+    state_tracker,
+};
+use crate::{errors::Error, genesis};
+
+/// Mutable worker state that we modify in the consensus worker task.
+///
+/// Unable to be shared across threads.  Any data we want to export we'll do
+/// through another handle.
+#[allow(unused)]
+pub struct WorkerState<D: Database> {
+    /// Consensus parameters.
+    params: Arc<Params>,
```

Comment:
This is a lib I don't think we should `Arc` stuff here. Maybe fine in binaries.
What if we
```suggestion
    params: &Params,
```
and let the user define if it needs `Arc` or no in the actual impl?

Reply:
We construct it as an Arc and pass it around everywhere as one.  Plus then we'd have to expose a lifetime param.